### PR TITLE
sources: add SCIM source

### PR DESCRIPTION
**Pull request by BeryJu**
Add SCIM source which provides endpoints for other applications to create/read/update/delete users and groups in authentik

### CODE REVIEW

**Comment by chaosgrid** [comment]
SCIM source would be great to have, is there an ETA for it? :)

File: website/docs/sources/scim.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+---
+title: SCIM
```
Comment (tanberry) [inline website/docs/sources/scim.md:2]:
```suggestion
title: SCIM source
```

File: website/docs/sources/scim.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+---
+
+The SCIM source allows other applications to directly create users and groups within authentik. SCIM is supported by applications such as Microsoft Azure AD, Google Workspace and Okta.
```
Comment (tanberry) [inline website/docs/sources/scim.md:5]:
```suggestion
The SCIM source allows other applications to directly create users and groups within authentik. SCIM provides predefined schema for users and groups, with a RESTful API, to enable automatic user provisioning and deprovisioning, SCIM is supported by applications such as Microsoft Azure AD, Google Workspace, and Okta.
```

File: website/docs/sources/scim.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+The base SCIM URL is in the format of `https://authentik.company/source/scim/<source-slug>/v2`. Authentication is done via Bearer tokens generated by authentik. When an SCIM source is created, a service account is created and a matching token.
```
Comment (tanberry) [inline website/docs/sources/scim.md:7]:
```suggestion
The base SCIM URL is in the format of `https://authentik.company/source/scim/<source-slug>/v2`. Authentication is done via Bearer tokens that are generated by authentik. When an SCIM source is created, a service account is created and a matching token is provided.
```

File: website/docs/sources/scim.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+## Supported Options & Resource types
+
+### `/v2/Users`
+
+Endpoint to list, create, patch and delete users.
```
Comment (tanberry) [inline website/docs/sources/scim.md:13]:
```suggestion
Endpoint to list, create, patch, and delete users.
```

**Review by tanberry — APPROVED** [review]
LGTM on the docs!

File: website/docs/sources/scim.md [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+### `/v2/Groups`
+
+Endpoint to list, create, patch and delete groups.
```
[the remark this answers is not in the document]
Comment (kensternberg-authentik) [reply website/docs/sources/scim.md:17]: Is there a reason to keep this PR on the books?

[the remark this answers is not in the document]
Comment (BeryJu) [reply website/docs/sources/scim.md:17]: yeah it still needs to be finished, this is currently the only thing we don't have from our own comparison table

**Pull request merged**